### ISSUE TITLE
Directory index is already disposed for Project (Disposed) MyProject

### ISSUE THREAD

**Issue by welearnednothing** [hsz/idea-gitignore#119]
I just got this after closing a project window and still having another project window open. I wasn't doing anything in IntelliJ at the time, I had just switched back to it and closed the project referenced in the error.

```
2015-05-04 12:24:07,610 [8641469]  ERROR - .roots.impl.DirectoryIndexImpl - Directory index is already disposed for Project (Disposed) MyProject 
java.lang.Throwable
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:115)
	at com.intellij.openapi.roots.impl.DirectoryIndexImpl.b(DirectoryIndexImpl.java:206)
	at com.intellij.openapi.roots.impl.DirectoryIndexImpl.getInfoForFile(DirectoryIndexImpl.java:163)
	at com.intellij.openapi.roots.impl.FileIndexBase.getInfoForFileOrDirectory(FileIndexBase.java:23)
	at com.intellij.openapi.roots.impl.ProjectFileIndexImpl.isInContent(ProjectFileIndexImpl.java:180)
	at com.intellij.psi.search.ProjectAndLibrariesScope.contains(ProjectAndLibrariesScope.java:43)
	at com.intellij.psi.search.GlobalSearchScope.accept(GlobalSearchScope.java:78)
	at com.intellij.util.indexing.FileBasedIndexImpl$14.convert(FileBasedIndexImpl.java:978)
	at com.intellij.util.indexing.FileBasedIndexImpl$14.convert(FileBasedIndexImpl.java:947)
	at com.intellij.util.indexing.FileBasedIndexImpl.a(FileBasedIndexImpl.java:920)
	at com.intellij.util.indexing.FileBasedIndexImpl.a(FileBasedIndexImpl.java:993)
	at com.intellij.util.indexing.FileBasedIndexImpl.getContainingFiles(FileBasedIndexImpl.java:877)
	at com.intellij.psi.search.FileTypeIndex.getFiles(FileTypeIndex.java:44)
	at mobi.hsz.idea.gitignore.IgnoreManager$4.run(IgnoreManager.java:288)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
	at com.intellij.util.Alarm$Request$1.run(Alarm.java:351)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:332)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:209)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:715)
	at java.awt.EventQueue.access$400(EventQueue.java:82)
	at java.awt.EventQueue$2.run(EventQueue.java:676)
	at java.awt.EventQueue$2.run(EventQueue.java:674)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:86)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:685)
	at com.intellij.ide.IdeEventQueue.e(IdeEventQueue.java:734)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:569)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:382)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:296)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:211)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:196)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:188)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:122)
2015-05-04 12:24:07,611 [8641470]  ERROR - .roots.impl.DirectoryIndexImpl - IntelliJ IDEA 14.1.2  Build #IU-141.713.2 
2015-05-04 12:24:07,611 [8641470]  ERROR - .roots.impl.DirectoryIndexImpl - JDK: 1.6.0_65 
2015-05-04 12:24:07,611 [8641470]  ERROR - .roots.impl.DirectoryIndexImpl - VM: Java HotSpot(TM) 64-Bit Server VM 
2015-05-04 12:24:07,611 [8641470]  ERROR - .roots.impl.DirectoryIndexImpl - Vendor: Apple Inc. 
2015-05-04 12:24:07,611 [8641470]  ERROR - .roots.impl.DirectoryIndexImpl - OS: Mac OS X 
2015-05-04 12:24:07,611 [8641470]  ERROR - .roots.impl.DirectoryIndexImpl - Last Action: ActivateChangesToolWindow 
2015-05-04 12:24:07,644 [8641503]   INFO - il.indexing.FileBasedIndexImpl - Rebuild requested for index filetypes 
java.lang.Throwable: java.lang.AssertionError: Already disposed: Project (Disposed) MyProject
	at com.intellij.util.indexing.FileBasedIndexImpl.scheduleRebuild(FileBasedIndexImpl.java:1215)
	at com.intellij.util.indexing.FileBasedIndexImpl.a(FileBasedIndexImpl.java:938)
	at com.intellij.util.indexing.FileBasedIndexImpl.a(FileBasedIndexImpl.java:993)
	at com.intellij.util.indexing.FileBasedIndexImpl.getContainingFiles(FileBasedIndexImpl.java:877)
	at com.intellij.psi.search.FileTypeIndex.getFiles(FileTypeIndex.java:44)
	at mobi.hsz.idea.gitignore.IgnoreManager$4.run(IgnoreManager.java:288)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
	at com.intellij.util.Alarm$Request$1.run(Alarm.java:351)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:332)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:209)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:715)
	at java.awt.EventQueue.access$400(EventQueue.java:82)
	at java.awt.EventQueue$2.run(EventQueue.java:676)
	at java.awt.EventQueue$2.run(EventQueue.java:674)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:86)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:685)
	at com.intellij.ide.IdeEventQueue.e(IdeEventQueue.java:734)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:569)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:382)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:296)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:211)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:196)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:188)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:122)
Caused by: java.lang.AssertionError: Already disposed: Project (Disposed) MyProject
	at com.intellij.openapi.components.impl.ComponentManagerImpl.getComponent(ComponentManagerImpl.java:192)
	at com.intellij.openapi.module.ModuleManager.getInstance(ModuleManager.java:42)
	at com.intellij.openapi.roots.impl.RootIndex.a(RootIndex.java:100)
	at com.intellij.openapi.roots.impl.RootIndex.<init>(RootIndex.java:84)
	at com.intellij.openapi.roots.impl.DirectoryIndexImpl.a(DirectoryIndexImpl.java:128)
	at com.intellij.openapi.roots.impl.DirectoryIndexImpl.getInfoForFile(DirectoryIndexImpl.java:168)
	at com.intellij.openapi.roots.impl.FileIndexBase.getInfoForFileOrDirectory(FileIndexBase.java:23)
	at com.intellij.openapi.roots.impl.ProjectFileIndexImpl.isInContent(ProjectFileIndexImpl.java:180)
	at com.intellij.psi.search.ProjectAndLibrariesScope.contains(ProjectAndLibrariesScope.java:43)
	at com.intellij.psi.search.GlobalSearchScope.accept(GlobalSearchScope.java:78)
	at com.intellij.util.indexing.FileBasedIndexImpl$14.convert(FileBasedIndexImpl.java:978)
	at com.intellij.util.indexing.FileBasedIndexImpl$14.convert(FileBasedIndexImpl.java:947)
	at com.intellij.util.indexing.FileBasedIndexImpl.a(FileBasedIndexImpl.java:920)
	... 24 more
2015-05-04 12:24:07,667 [8641526]  ERROR - .roots.impl.DirectoryIndexImpl - Directory index is already disposed for Project (Disposed) MyProject 
java.lang.Throwable
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:115)
	at com.intellij.openapi.roots.impl.DirectoryIndexImpl.b(DirectoryIndexImpl.java:206)
	at com.intellij.openapi.roots.impl.DirectoryIndexImpl.getInfoForFile(DirectoryIndexImpl.java:163)
	at com.intellij.openapi.roots.impl.FileIndexBase.getInfoForFileOrDirectory(FileIndexBase.java:23)
	at com.intellij.openapi.roots.impl.ProjectFileIndexImpl.isInContent(ProjectFileIndexImpl.java:180)
	at com.intellij.psi.search.ProjectAndLibrariesScope.contains(ProjectAndLibrariesScope.java:43)
	at com.intellij.psi.search.GlobalSearchScope.accept(GlobalSearchScope.java:78)
	at com.intellij.util.indexing.FileBasedIndexImpl$14.convert(FileBasedIndexImpl.java:978)
	at com.intellij.util.indexing.FileBasedIndexImpl$14.convert(FileBasedIndexImpl.java:947)
	at com.intellij.util.indexing.FileBasedIndexImpl.a(FileBasedIndexImpl.java:920)
	at com.intellij.util.indexing.FileBasedIndexImpl.a(FileBasedIndexImpl.java:993)
	at com.intellij.util.indexing.FileBasedIndexImpl.getContainingFiles(FileBasedIndexImpl.java:877)
	at com.intellij.psi.search.FileTypeIndex.getFiles(FileTypeIndex.java:44)
	at mobi.hsz.idea.gitignore.IgnoreManager$4.run(IgnoreManager.java:288)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
	at com.intellij.util.Alarm$Request$1.run(Alarm.java:351)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:332)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:209)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:715)
	at java.awt.EventQueue.access$400(EventQueue.java:82)
	at java.awt.EventQueue$2.run(EventQueue.java:676)
	at java.awt.EventQueue$2.run(EventQueue.java:674)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:86)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:685)
	at com.intellij.ide.IdeEventQueue.e(IdeEventQueue.java:734)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:569)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:382)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:296)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:211)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:196)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:188)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:122)
2015-05-04 12:24:07,668 [8641527]  ERROR - .roots.impl.DirectoryIndexImpl - IntelliJ IDEA 14.1.2  Build #IU-141.713.2 
2015-05-04 12:24:07,668 [8641527]  ERROR - .roots.impl.DirectoryIndexImpl - JDK: 1.6.0_65 
2015-05-04 12:24:07,668 [8641527]  ERROR - .roots.impl.DirectoryIndexImpl - VM: Java HotSpot(TM) 64-Bit Server VM 
2015-05-04 12:24:07,668 [8641527]  ERROR - .roots.impl.DirectoryIndexImpl - Vendor: Apple Inc. 
2015-05-04 12:24:07,668 [8641527]  ERROR - .roots.impl.DirectoryIndexImpl - OS: Mac OS X 
2015-05-04 12:24:07,668 [8641527]  ERROR - .roots.impl.DirectoryIndexImpl - Last Action: ActivateChangesToolWindow 
2015-05-04 12:24:07,671 [8641530]  ERROR - til.concurrency.QueueProcessor - Already disposed: Project (Disposed) MyProject 
java.lang.AssertionError: Already disposed: Project (Disposed) MyProject
	at com.intellij.openapi.components.impl.ComponentManagerImpl.getPicoContainer(ComponentManagerImpl.java:282)
	at com.intellij.openapi.project.impl.ProjectImpl.getStateStore(ProjectImpl.java:187)
	at com.intellij.openapi.project.impl.ProjectImpl.getBaseDir(ProjectImpl.java:238)
	at mobi.hsz.idea.gitignore.lang.kind.FossilLanguage.getOuterFile(FossilLanguage.java:74)
	at mobi.hsz.idea.gitignore.IgnoreManager$4.run(IgnoreManager.java:298)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
	at com.intellij.util.Alarm$Request$1.run(Alarm.java:351)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:332)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:209)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:715)
	at java.awt.EventQueue.access$400(EventQueue.java:82)
	at java.awt.EventQueue$2.run(EventQueue.java:676)
	at java.awt.EventQueue$2.run(EventQueue.java:674)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:86)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:685)
	at com.intellij.ide.IdeEventQueue.e(IdeEventQueue.java:734)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:569)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:382)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:296)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:211)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:196)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:188)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:122)
```

**Comment by hsz** [comment]
Thank you for reporting. The issue has been already reported in #109 and fixed.

It will be pushed with the next release.

**Comment by welearnednothing** [comment]
Ah shoot, I didn't look through the fixed issues. Sorry for the noise!

**Comment by hsz** [comment]
No problem. Better to have few duplicated issues than none of them. ;-)

**Comment by hsz** [comment]
Fix has ben released just right now with `v1.1.2`
